app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -0,0 +1,151 @@
+package com.example.drawingactivity
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+class DialogUtils(
```

Comment:
like Nic said, should be an object not a class, also shouldn't hold variables for context or ViewModel because its static, does mean that you can store consts for example a max Char length